Institutions

awolfe76 opened the pull request:
- updates institutions listing page to be a bit more friendly
  - the `status.message` is color coded, green for completed, red for not started and cancelled, and blue for in progress
  - the description text is modified based on the `status.code` too
- adds [moment.js](http://momentjs.com/) to handle time rendering
- adds buttons based on status, with the refile button always present but smaller
- adds column to grid with placeholder text, we can add help text or other messages there
- would like to know if you have thoughts/suggestions on the [`renderTiming`](https://github.com/cfpb/hmda-platform-ui/compare/institutions?expand=1#diff-86bed846ff53cc9ec2c51bcb085621d4R6) and [`renderStatus`](https://github.com/cfpb/hmda-platform-ui/compare/institutions?expand=1#diff-86bed846ff53cc9ec2c51bcb085621d4R26) functions

#### Screenshot
![hmda-fis](https://cloud.githubusercontent.com/assets/2932572/20633989/18238c1e-b31a-11e6-8f44-f8f4a5a7edd4.png)

awolfe76 committed:
add moment.js to handle time outputs

awolfe76 committed:
rendering institutions in a cleaner way, along with relevant times of submissions

awolfe76 committed:
clean up on how the timing and status are rendered

awolfe76 committed:
rework layout of institutions, clean up functions a bit

awolfe76 committed:
remove link from status now that there are buttons

wpears commented:
Cool changes.

I think there needs to be more hierarchy/spacing within a bank's filings though (see attached). Also, I think the filing period absolutely needs to be called out in the filing header and perhaps the repetition of the institution name is redundant.

Also, looks like the `start` and `end` fields are 0 when the filing hasn't been started and this isn't being accurately dealt with in `renderTiming`

<img width="1046" alt="screen shot 2016-11-28 at 3 11 51 pm" src="https://cloud.githubusercontent.com/assets/1558033/20690040/20f5d1ca-b57d-11e6-86c2-8fe7d356011d.png">

awolfe76 committed:
fix for rendering timing when code is 1, not-started

awolfe76 committed:
update status text to avoid repetition of FI name and reorganize buttons to be better grouped with FI content

awolfe76 committed:
better spacing for status

awolfe76 committed:
add ID

awolfe76 commented:
> I think there needs to be more hierarchy/spacing within a bank's filings though (see attached).

We wouldn't have this case though would we? When someone logs in they would only see the current year. For now, this is only 2017. In the future we would provide some way to go to previous years, maybe a drop-down. I think https://github.com/cfpb/hmda-platform-ui/issues/269 helps here too.

> Also, looks like the start and end fields are 0 when the filing hasn't been started and this isn't being accurately dealt with in renderTiming

Fixed in https://github.com/cfpb/hmda-platform-ui/pull/268/commits/8b00fc953ddb9f40c33e2fc93db3f3ec660467d7

> perhaps the repetition of the institution name is redundant.

Updated in https://github.com/cfpb/hmda-platform-ui/pull/268/commits/81b8276a7c99050241a73f31370403423a9709cd, but also think that we'll improve this even more very soon.

Also added the FI ID (Name - ID), we can figure out what the best ID is later to add to the API.
And changed the position of the buttons, this seems like a better flow.

#### Updated Screenshot

![hmda-fi](https://cloud.githubusercontent.com/assets/2932572/20756228/dc2f5444-b6df-11e6-81ca-7be531331493.png)

awolfe76 committed:
only render refile if status is not started
